perfdash/metrics-downloader.go

```diff
@@ -279,6 +285,109 @@ func (b *GCSMetricsBucket) GetFile(job string, buildNumber int, path string) ([]
 	return data, nil
 }
 
+// S3MetricsBucket that creates a AWS S3 client to fetch data.
+type S3MetricsBucket struct {
+	client  *s3.S3
+	bucket  *string
+	logPath string
+}
+
+// NewS3MetricsBucket creates a new S3MetricsBucket.
+func NewS3MetricsBucket(bucket, path, region string) (MetricsBucket, error) {
```

Comment:
just a nit, if it has a path it isn't a bucket anymore. Since you're specifying a path, this object is more of an S3MetricsDirectory.

Reply:
I'd support renaming `path` to `pathPrefix` or just `prefix`

Reply:
Renamed `path` to `pathPrefix`
I totally agree with the naming though in regards to "if it has a path, it isn't a bucket anymore." I would be open to refactoring in the future if that's something we're looking for.